main.go

```diff
@@ -0,0 +1,242 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/bitrise-io/go-steputils/stepconf"
+	"github.com/bitrise-io/go-steputils/tools"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/google/go-querystring/query"
+	"github.com/pkg/errors"
+)
+
+const (
+	baseURL                   = "https://api.bitrise.io/v0.1"
+	buildTypeTag              = "tag"
+	buildTypePush             = "push"
+	buildTypeManual           = "manual"
+	buildTypePullRequest      = "pull-request"
+	envKeyPreviousBuildStatus = "PREVIOUS_BUILD_STATUS"
+	envKeyBuildStatusChanged  = "BUILD_STATUS_CHANGED"
+	statusTextSuccessfulBuild = "success"
+)
+
+type config struct {
+	AppSlug             string          `env:"BITRISE_APP_SLUG,required"`
+	BuildSlug           string          `env:"BITRISE_BUILD_SLUG,required"`
+	BuildStatus         string          `env:"BITRISE_BUILD_STATUS,required"`
+	PreviousBuildStatus string          `env:"PREVIOUS_BUILD_STATUS"`
+	AccessToken         stepconf.Secret `env:"access_token,required"`
+}
+
+func (cfg config) getBuild() (build, error) {
+	url := baseURL + "/apps/" + cfg.AppSlug + "/builds/" + cfg.BuildSlug
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return build{}, err
+	}
+	req.Header.Add("accept", "application/json")
+	req.Header.Add("Authorization", string(cfg.AccessToken))
+
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
I thought we earlier we agreed to not use the default http client due to timeout issues.

Reply:
retry could be useful on failure.